### ISSUE TITLE
JSONTransformer method not called

### ISSUE THREAD

**Issue by adil-hussain-84** [Mantle/Mantle#700]
I'm using version 2.0.7 of the Mantle library (the latest at the time of writing this issue). I have a JSON object that I'd like to map to a typed object. And within that JSON object there's a JSON array of objects that I'd also like each to be mapped to a typed object.

The header for my outer typed object is as follows:

```
#import <Foundation/Foundation.h>
#import <Mantle/Mantle.h>

@interface OuterObject : MTLModel <MTLJSONSerializing>

@property(nonatomic, assign) int resultCount;
@property(nonatomic, strong) NSArray *results;

@end
```

The implementation for the outer typed object is as follows:

```
#import "OuterObject.h"
#import "InnerObject.h"

@implementation OuterObject

+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    return @{
        @"resultCount" : @"resultCount",
        @"results" : @"results"
    };
}

+ (NSValueTransformer *)resultsJSONTransformer {
    return [MTLJSONAdapter arrayTransformerWithModelClass:InnerObject.class];
}
```

The header for the inner object is as follows:

```
#import <Foundation/Foundation.h>
#import <Mantle/Mantle.h>

@interface InnerObject : MTLModel <MTLJSONSerializing>

@property(nonatomic, strong) NSString *keyA;
@property(nonatomic, strong) NSString *keyB;

@end
```

The implementation for the inner object is as follows:

```
#import "InnerObject.h"

@implementation InnerObject

+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    return @{
        @"keyA" : @"keyA",
        @"keyB" : @"keyB"
    };
}

@end
```

I'm attempting to parse an instance of `NSDictionary` to an instance of `OuterObject` as follows:

```
NSMutableDictionary *innerObjectDictionary = [[NSMutableDictionary alloc] init];
[innerObjectDictionary setValue:@"valueA" forKey:@"keyA"];
[innerObjectDictionary setValue:@"valueB" forKey:@"keyB"];

NSArray *resultsArray = [NSArray arrayWithObject:innerObjectDictionary];

NSMutableDictionary *outerObjectDictionary = [[NSMutableDictionary alloc] init];
[outerObjectDictionary setValue:[NSNumber numberWithInt:1] forKey:@"resultCount"];
[outerObjectDictionary setValue:resultsArray forKey:@"results"];

OuterObject *outerObject = [OuterObject modelWithDictionary:outerObjectDictionary error:&err];
```

Unfortunately the `resultsJSONTransformer` method is never called and thus the items in the list are not transformed from instances of `NSDictionary` to instances of `InnerObject`. Can anyone see why? I'm been at it for a while now and totally out of ideas :-(

**Comment by dcaunt** [comment]
When you use the `modelWithDictionary:error` method, you should provide a dictionary mapping of your model's property names to their values, for what is really a constructor for the model. 

JSON transformers are only invoked when you use `MTLJSONAdapter` to transform JSON to models, i.e. [`modelOfClass:fromJSONDictionary:error:`](https://github.com/Mantle/Mantle/blob/a16d18f635b3ba1ffb01d27d4393ee97cadf4f34/Mantle/MTLJSONAdapter.h#L115). Internally, `MTLJSONAdapter` uses the JSON transformers and JSON key mappings to create dictionaries with keys and values for the `modelWithDictionary:error:` method's dictionary parameter.

In short: 
```
OuterObject *outerObject = [MTLJSONAdapter modelOfClass:OuterObject.class fromJSONDictionary:outerObjectDictionary error&err]
```

**Comment by dcaunt** [comment]
Closing this due to inactivity, please feel free to reopen or create a new issue if you have further questions.
